File: MasterOfCode/ServicesModule/src/main/java/Sockets/WebSocket.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Sockets;

import java.util.List;
import java.util.Map;
import javax.websocket.Session;

/**
 *
 * @author mikerooijackers
 */
public class WebSocket {
    private List<Session> spectatorPeers;
    private Map<Long, Session> userPeers;
    private Map<Long, Session> participantTeam;
    
    /**
     *
     * @param userID
     * @param teamID
     * @param client
     * @param message
     */
    public void OnMessage(Long userID, Long teamID, Session client, String message) {
        
    }
    
    /**
     *
     * @param userID
     * @param teamID
     * @param client
     */
    public void OnClose(Long userID, Long teamID, Session client) {
        
    }
    
    /**
     *
     * @param userID
     * @param teamID
     * @param client
     */
    public void OnOpen(Long userID, Long teamID, Session client) {
        
    }
    
    /**
     *
     * @param t
     */
    public void OnError(Throwable t) {
        
    }
    
    /**
     *
     * @param UserID
     * @param message
     */
    public void SendMessageToUser(long UserID, Object message) {
        
    }
    
    /**
     *
     * @param TeamID
     * @param message
     */
    public void SendMessageToTeam(long TeamID, Object message) {
        
    }
    
    /**
     *
     * @param message
     */
    public void SendMessageToSpectator(Object message) {
        
    }
    
    /**
     *
     * @param message
     */
    public void SendMessageToParticipant(Object message) {
        
    }
}

File: MasterOfCode/AudienceClientModule/public_html/Scripts/websocket.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

// Create a websocket
var socket = new WebSocket("ws://");

// Check for a websocket 
$(document).ready(function () {
    if (!("WebSocket" in window)) {
        $('#chatLog, input, button, #examples').fadeOut("fast");
        $('<p>Oh no, you need a browser that supports WebSockets.</p>').appendTo('#container');
    } else {

        //The user has WebSockets
        connect();

        function connect() {
            //the connect function code is below
            try {

                var socket;
                var host = "ws://";
                var socket = new WebSocket(host);

                message('<p class="event">Socket Status: ' + socket.readyState);

                socket.onopen = function () {
                    message('<p class="event">Socket Status: ' + socket.readyState + ' (open)');
                };

                socket.onmessage = function (msg) {
                    message('<p class="message">Received: ' + msg.data);
                };

                socket.onclose = function () {
                    message('<p class="event">Socket Status: ' + socket.readyState + ' (Closed)');
                };

            } catch (exception) {
                message('<p>Error' + exception);
            }
        }
    }
});

